internal/install/recipe_installer.go

```diff
@@ -138,6 +144,17 @@ func (i *RecipeInstaller) discoverAndRun(ctx context.Context) error {
 	return i.guidedInstall(ctx, m)
 }
 
+func (i *RecipeInstaller) assertDiscoveryValid(ctx context.Context, m *types.DiscoveryManifest) error {
+	if m.OS == "" {
+		return fmt.Errorf("failed to identify a valid operating system")
```

Comment:
We've started putting errors in the `internal/install/types` package.  What do you think about moving these over there and then just returning the particular error here?

Reply:
I'm gonna refrain from doing this at this time